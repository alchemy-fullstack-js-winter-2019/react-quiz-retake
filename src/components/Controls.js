import React from 'react';
import PropTypes from 'prop-types';

function Controls({ title, subtitle, onSubmit, onChange }) {
  return (
    <form onSubmit={onSubmit}>
    <input type="text" name="title" vslue={title} onChange={onChange} />
    <input type="text" name="subtitle" vslue={subtitle} onChange={onChange} />
    <button type="submit">Save</button>
    </form>
  );
}

Controls.propTypes = {
  title: propTypes.string.isRequired,
  subtitle: propTypes.string
};

export default Controls;
