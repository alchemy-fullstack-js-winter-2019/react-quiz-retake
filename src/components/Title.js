import React from 'react';
import PropTypes from 'prop-types';

function Title({ title }) {
  return (
    <p>{title}</p>
  );
}
Title.propTypes = {
  title: PropTypes.string
};

export default Title;
