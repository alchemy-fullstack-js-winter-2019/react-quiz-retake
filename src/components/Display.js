import React from 'react';
import PropTypes from 'prop-types';
import Title from './Title';

function Display({ titles }) {
  const listOfTitles = titles.map(title => {
    return (
      <li key={title.name}>
        <Title title={title} value={title} />
      </li>
    );
  });
  return (
    <ul>
      {listOfTitles}
    </ul>
  );
}
Display.propTypes = {
  titles: PropTypes.array.isRequired
};

export default Display;

