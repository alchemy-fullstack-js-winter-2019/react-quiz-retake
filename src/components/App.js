import React, { Component } from 'react';
import Controls from './Controls';
import Display from './Display';

export default class App extends Component {

  state = {
    title: '',
    subtitle: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  render() {
    const {
      title,
      subtitle
    }
    = this.state;

    return (
    <>
    <h1>REACT QUIZ RETAKE</h1>

    <Controls
      title={title}
      subtitle={subtitle}
      onChange={this.handleChange}
    />
  

    <Display 
      title={title}
      subtitle={subtitle}
    />

    </>
    );
  }
}
