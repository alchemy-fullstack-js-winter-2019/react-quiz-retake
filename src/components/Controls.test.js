import React from 'react';
import renderer from 'react-test-renderer';
import { Controls } from '../components/Controls';

describe('Controls test', () => {
  it('matches a snapshot', () => {
    const controls = jest.fn();
    const tree = renderer.create(
      <Controls controls={controls} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

