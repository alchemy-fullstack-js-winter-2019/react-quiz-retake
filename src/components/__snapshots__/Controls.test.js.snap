// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls matches a snapshot 1`] = `
<form>
  <label>
    Title Text:
  </label>
  <input
    name="title"
    type="text"
  />
  <label>
    Subtitle Text:
  </label>
  <input
    name="subtitle"
    type="text"
  />
</form>
`;
